test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+                    expect(typeof value).to.eql('number');
+                });
+                expect(_.min.apply(_, params)).to.eql(-3);
+            });
+        });
+
+
+        describe('minBy', function () {
+            
+            describe('minBy - Array', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy1();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(params[0] instanceof Array).to.eql(true);
+                });
+
+                it('should return min value in Array', function () {
+                    expect(_.min.apply(_, params)).to.eql(-1);
+                });
+            });
+
+            /*
+            describe('minBy - Function', function () {
+                it('Function', function () {
+                });
+            });
+            //I have no idea how to do it. Probably minBy doesn't support function. Help!
+            */
+
+            describe('minBy - Object', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy3();
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(2);
+                    expect(params[0] instanceof Object).to.eql(true);
+                    expect(typeof params[1]).to.eql('string');
+                });
+
+                it('should return min value from Object', function () {
+                    expect(_.min.apply(_, params)).to.eql({ Name: 'Jack', Gold: 1500 });
+                });
+            });
+
+            describe('minBy - String', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.minBy4();
+                });
+                
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(typeof params[0]).to.eql('string');
+                });
+                
+                it('should return min value from string', function () {
+                    expect(_.min.apply(_, params)).to.eql('e');
+                });
+            });
+        });
+
+        describe('multiply', function () {
+
+            var val1;
+            var val2;
+            var result1;
+            var params = datasets.multiply();
+
+            beforeEach(function () {
+                val1 = Math.round(Math.random() * 10);
+                val2 = Math.round(Math.random() * 10);
+                result1  = val1 * val2;
+            });
+
+            it('should multiply two random number', function () {
+                expect(params).to.have.length(2);
+                expect(_.multiply.apply(_, datasets.multiply(val1, val2))).to.eql(result1);
```

Comment:
How does that test knowledge of multiply function at all?

Maybe instead trying to do some magic you should provide fixed numbers and expect user to give you an answer based on numbers you provided for test case?